laravel-live-coding/tests/Unit/UserControllerTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Tests\Unit;
+
+use App\Repositories\UserRepository;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+class UserControllerTest extends TestCase
+{
+    use RefreshDatabase;
+
+    public function testIndexReturnsCorrectResponse()
+    {
+        // Arrange
+        $mockUserData = [
+            [
+                'name' => 'Dima',
+                'email' => '@gmail.com',
+                'phone' => '+38000000000',
+                'photo' => 'https://d3jqtupnzefbtn.cloudfront.net/andersenlab/new-andersensite/testimonials/60x60/without-photo.png',
+                'wallets' => [
+                    [
+                        'currency' => 'USD',
+                        'amount' => 12321314124134e312321,
+                    ],
+                    [
+                        'currency' => 'EUR',
+                        'amount' => 12321314124134e312321,
+                    ],
+                ],
+            ],
+            [
+                'name' => 'Mariya',
+                'email' => '@gmail.com',
+                'phone' => '+3800312300000',
+                'photo' => 'https://d3jqtupnzefbtn.cloudfront.net/andersenlab/new-andersensite/testimonials/60x60/without-photo.png',
+                'wallets' => [
+                    [
+                        'currency' => 'USD',
+                        'amount' => 12321314124134e312321,
+                    ],
+                    [
+                        'currency' => 'EUR',
+                        'amount' => 12321314124134e312321,
+                    ],
+                ],
+            ],
+        ];
+
+        $mockUserRepository = $this->createMock(UserRepository::class);
+        $mockUserRepository->expects($this->once())
+            ->method('getAllUsersWithWallets')
+            ->willReturn($mockUserData);
+
+        $this->app->instance(UserRepository::class, $mockUserRepository);
+
+        // Act
+        $response = $this->get('/api/user');
+
+        // Assert
+        $response->assertStatus(200)
```

Comment:
You wrote the **functional** test without any sense.

You created a mock of which one you want to test))) 
Just read your code you **create mock** and tell him what he **must return.**
And after that, you **assertJsonStructure** which you put in the **mock**) so you **do not check your code**, you check **your mock structure**

Mock will be useful when you have some remote service, from another app, where you can't predicted what you can resive

in this test try to work with sqlLite and remove ->willReturn($mockUserData); and it will work!)